packages/server/graphql/mutations/addPokerTemplateScaleValue.ts

```diff
@@ -0,0 +1,101 @@
+import {GraphQLID, GraphQLNonNull} from 'graphql'
+import {SubscriptionChannel} from 'parabol-client/types/constEnums'
+import getRethink from '../../database/rethinkDriver'
+import {getUserId, isTeamMember} from '../../utils/authorization'
+import publish from '../../utils/publish'
+import standardError from '../../utils/standardError'
+import AddTemplateScaleInput from '../types/AddTemplateScaleInput'
+import AddPokerTemplateScaleValuePayload from '../types/AddPokerTemplateScaleValuePayload'
+import {
+  validateColorValue,
+  validateScaleLabel,
+  validateScaleValue,
+  validateScaleLabelValueUniqueness
+} from './helpers/validateScaleValue'
+
+const addPokerTemplateScaleValue = {
+  description: 'Add a new scale value for a scale in a poker template',
+  type: new GraphQLNonNull(AddPokerTemplateScaleValuePayload),
+  args: {
+    scaleId: {
+      type: new GraphQLNonNull(GraphQLID)
+    },
+    scaleValue: {
+      type: new GraphQLNonNull(AddTemplateScaleInput)
+    }
+  },
+  async resolve(_source, {scaleId, scaleValue}, {authToken, dataLoader, socketId: mutatorId}) {
+    const r = await getRethink()
+    const now = new Date()
+    const operationId = dataLoader.share()
+    const subOptions = {operationId, mutatorId}
+    const viewerId = getUserId(authToken)
+
+    // AUTH
+    const existingScale = await r
+      .table('TemplateScale')
+      .get(scaleId)
+      .run()
+    if (!existingScale || existingScale.removedAt) {
+      return standardError(new Error('Did not find an active scale'), {userId: viewerId})
+    }
+    if (!isTeamMember(authToken, existingScale.teamId)) {
+      return standardError(new Error('Team not found'), {userId: viewerId})
+    }
+
+    // VALIDATION
+    const {color, label, value, isSpecial} = scaleValue
+    if (!validateColorValue(color)) {
+      return standardError(new Error('Invalid scale color'), {userId: viewerId})
+    }
+    if (!validateScaleLabel(label)) {
+      return standardError(new Error('Invalid scale label'), {userId: viewerId})
+    }
+    if (!validateScaleValue(value, !!isSpecial)) {
+      return standardError(new Error('Invalid scale value'), {userId: viewerId})
+    }
+
+    await r
+      .table('TemplateScale')
+      .get(scaleId)
+      .update(
+        (row) => ({
+          values: row('values').append(scaleValue),
+          updatedAt: now
+        }),
+        {returnChanges: true}
+      )
```

Comment:
i see you started returning changes but then broke it into 2.

on thing line, you can add `('changes')(0)('new_val').default(null)`
if it's null, then that means the update didn't occur.

since atomicity is at play here, it's probably worth doing it all at once